common/network-common/src/main/java/org/apache/spark/network/util/ByteArrayReadableChannel.java

```diff
@@ -19,44 +19,44 @@
 
 import java.io.IOException;
 import java.nio.ByteBuffer;
+import java.nio.channels.ClosedChannelException;
 import java.nio.channels.ReadableByteChannel;
 
 import io.netty.buffer.ByteBuf;
 
 public class ByteArrayReadableChannel implements ReadableByteChannel {
   private ByteBuf data;
+  private boolean closed;
```

Comment:
Not sure, does this need to be volatile? I'm not sure if there are thread safety concerns here.

Reply:
streams / channels are not thread-safe.